src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/HalfConverter.cs

```diff
@@ -0,0 +1,228 @@
+            {
+#if NET8_0_OR_GREATER
+                const byte Quote = JsonConstants.Quote;
+                Span<byte> buffer = stackalloc byte[MaxFormatLength + 2];
+#else
+                const char Quote = (char)JsonConstants.Quote;
+                Span<char> buffer = stackalloc char[MaxFormatLength + 2];
+#endif
+                buffer[0] = Quote;
+                Format(buffer.Slice(1), value, out int written);
+
+                int length = written + 2;
+                buffer[length - 1] = Quote;
+                writer.WriteRawValue(buffer.Slice(0, length));
+            }
+            else if ((JsonNumberHandling.AllowNamedFloatingPointLiterals & handling) != 0)
+            {
+                WriteFloatingPointConstant(writer, value);
+            }
+            else
+            {
+                WriteCore(writer, value);
+            }
+        }
+
+        private static bool TryGetFloatingPointConstant(ref Utf8JsonReader reader, out Half value)
+        {
+            Span<byte> buffer = stackalloc byte[MaxFormatLength];
+            int written = reader.CopyValue(buffer);
+
+            return JsonReaderHelper.TryGetFloatingPointConstant(buffer.Slice(0, written), out value);
+        }
+
+        private static void WriteFloatingPointConstant(Utf8JsonWriter writer, Half value)
+        {
+            if (Half.IsNaN(value))
+            {
+                writer.WriteNumberValueAsStringUnescaped(JsonConstants.NaNValue);
+            }
+            else if (Half.IsPositiveInfinity(value))
+            {
+                writer.WriteNumberValueAsStringUnescaped(JsonConstants.PositiveInfinityValue);
+            }
+            else if (Half.IsNegativeInfinity(value))
+            {
+                writer.WriteNumberValueAsStringUnescaped(JsonConstants.NegativeInfinityValue);
+            }
+            else
+            {
+                WriteCore(writer, value);
+            }
+        }
+
+        // Half.TryFormat/TryParse(ROS<byte>) are not available on .NET 7
+        // we need to use Half.TryFormat/TryParse(ROS<char>) in that case.
+        private static bool TryParse(
+#if NET8_0_OR_GREATER
+            ReadOnlySpan<byte> buffer,
+#else
+            ReadOnlySpan<char> buffer,
+#endif
+            out Half result)
+        {
+            bool success = Half.TryParse(buffer, NumberStyles.Float | NumberStyles.AllowThousands, CultureInfo.InvariantCulture, out result);
+
+            // Half.TryParse is more lax with floating-point literals than other S.T.Json floating-point types
+            // e.g: it parses "naN" successfully. Only succeed with the exact match.
+#if NET8_0_OR_GREATER
+            ReadOnlySpan<byte> NaN = JsonConstants.NaNValue;
+            ReadOnlySpan<byte> PositiveInfinity = JsonConstants.PositiveInfinityValue;
+            ReadOnlySpan<byte> NegativeInfinity = JsonConstants.NegativeInfinityValue;
+#else
+            const string NaN = "NaN";
```

Comment:
I was able to remove these `const string`s in the latest commit, since we are passing the bytes in all tfms, we can just compare with the existing utf8 JsonConstants.